crates/source_file/src/source_file.rs

```diff
@@ -0,0 +1,966 @@
+//! This top level documentation details the algorithms and terminology behind
```

Comment:
Spent a lot of time on this documentation. I don't think you need to read it in detail unless you really want to know how `offset()` and `source_location()` work